### PR TITLE
Re-do the documentation

### LINKED ISSUE #160: Refresh: Update all documentation and examples with respect to the refresh

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #160 #202

# Summary

- All documentation about the tooling has been updated. There should be no references to the old turnkey tool syntax.
- Brand new top-level README.md.
  - The stated purpose of the repo has been greatly simplified: this is a turnkey way to run ONNX ecosystem tools, period.
  - The most important outcome of this document is that users have an idea of how to drive the CLI's unique syntax.
- The tutorials under examples/cli have been replaced by a much simpler tree of readme files and docstrings.
- The tools user guide is now only used to explain syntax, concepts, and environment variables.
  - Our expectation is that users should do most of their learning through `-h` menus.
  - `Tool` docstrings now appear in the `turnkey TOOL -h` printout, which has greatly expanded the information provided to users.
- Rev'd the version number to `3.0.0` in anticipation of The Big Merge.

# TODO

- [x] Make a nice animated gif for the front readme.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,165 +1,92 @@
...
-                Class:          BertModel (<class 'transformers.models.bert.modeling_bert.BertModel'>)
-                Location:       /home/jfowers/turnkeyml/models/transformers/bert.py, line 23
-                Parameters:     109,482,240 (417.64 MB)
-                Input Shape:    'attention_mask': (1, 128), 'input_ids': (1, 128)
-                Hash:           bf722986
-                Build dir:      /home/jfowers/.cache/turnkey/bert_bf722986
-                Status:         Successfully benchmarked on AMD Ryzen 9 7940HS w/ Radeon 780M Graphics (ort v1.15.1) 
-                                Mean Latency:   44.168  milliseconds (ms)
-                                Throughput:     22.6    inferences per second (IPS)
-```
```
Comment (danielholanda) [inline README.md:66]: This will get much easier to maintain without those.

File: docs/release_notes.md [hunk 2]
```diff
@@ -1,179 +0,0 @@
-# Release Notes
```
Comment (danielholanda) [inline docs/release_notes.md:1]: Should we simply start [automatically generating release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)?

File: src/turnkeyml/version.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "2.0.3"
+__version__ = "3.0.0"
```
Comment (danielholanda) [inline src/turnkeyml/version.py:1]: Woohoo!

File: src/turnkeyml/tools/management_tools.py [hunk 4]
```diff
@@ -216,23 +234,28 @@ def run(
 
 class ModelsLocation(ManagementTool):
     """
-    Prints the location of the turnkeyml built in models corpus.
+    Prints the location of the turnkeyml built in models corpora.
+
+    This is especially useful for when turnkey was installed from PyPI
+    with `pip install turnkeyml`. Example usage in this context:
+        models=$(turnkey models-location --quiet)
```
Comment (danielholanda) [inline src/turnkeyml/tools/management_tools.py:241]: Outside the scope, but I really with there was an easier way to do this.
(e.g. merging model directories during installation)

File: src/turnkeyml/cli/cli.py [hunk 5]
```diff
@@ -68,7 +43,7 @@ def _check_extension(
     if not extension:
         close_matches = get_close_matches(file_name, tool_names)
         if close_matches:
-            # Mispelled tool names can be picked up as input files, so we check
+            # Misspelled tool names can be picked up as input files, so we check
```
Comment (danielholanda) [inline src/turnkeyml/cli/cli.py:46]: 😂

**Review by danielholanda — APPROVED** [review]
Finally documentation that is digestible!

File: docs/release_notes.md (hunk 2 above)
Comment (jeremyfowers) [reply docs/release_notes.md:1]: That's what we switched to for the last couple releases, which is why I'm just getting rid of this file now.

File: README.md (hunk 1 above)
Comment (jeremyfowers) [reply README.md:66]: Yeah that was not fun!

File: README.md [hunk 6]
```diff
@@ -1,165 +1,92 @@
...
-
-The [installation guide](https://github.com/onnx/turnkeyml/blob/main/docs/install.md), [tutorials](https://github.com/onnx/turnkeyml/tree/main/examples/cli), and [user guide](https://github.com/onnx/turnkeyml/blob/main/docs/tools_user_guide.md) have everything you need to know.
+## Demo
 
-## Use Cases
+Here's `turnkey` in action: BERT-Base is exported from PyTorch to ONNX using `torch.onnx.export`, optimized for inference with `onnxruntime`, and converted to float16 with `onnxmltools`:
 
-TurnkeyML is designed to support the following use cases. Of course, it is also quite flexible, so we are sure you will come up with some use cases of your own too.
+![Basic Demo Video](img/basic_demo.gif)
```
[the remark this answers is not in the document]
Comment (jeremyfowers) [reply README.md:19]: Added! Seems to work too :)

https://github.com/onnx/turnkeyml/blob/jfowers/docs/README.md

File: src/turnkeyml/tools/management_tools.py (hunk 4 above)
Comment (jeremyfowers) [reply src/turnkeyml/tools/management_tools.py:241]: +1!

File: src/turnkeyml/version.py (hunk 3 above)
Comment (ramkrishna2910) [reply src/turnkeyml/version.py:1]: Nice!!

**Review by ramkrishna2910 — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,165 +1,92 @@
...
+The easiest way to learn more about `turnkey` is to explore the help menu with `turnkey -h`. To learn about a specific tool, run `turnkey <tool name> -h`, for example `turnkey export-pytorch -h`.
 
-Now you want to see the fp16 model running on your Nvidia GPU with the Nvidia TensorRT runtime:
+We also provide the following resources:
 
-```
-> turnkey bert.py --sequence export optimize-fp16 --device nvidia --runtime tensorrt
-```
+- [Installation guide](https://github.com/onnx/turnkeyml/blob/main/docs/install.md): how to install from source, set up Slurm, etc.
+- [User guide](https://github.com/onnx/turnkeyml/blob/main/docs/tools_user_guide.md): explains `turnkey's` concepts, including the syntax for making your own tool sequence.
```
Comment (vgodsoe) [inline README.md:61]:
```suggestion
- [User guide](https://github.com/onnx/turnkeyml/blob/main/docs/tools_user_guide.md): explains the concepts of `turnkey`, including the syntax for making your own tool sequence.
```
Just avoids possessive on a key name.

**Review by vgodsoe — COMMENTED** [review]
Super minor suggestions

**Pull request merged**